app/src/main/java/com/example/cooky/ui/general/content/GeneralContentViewModel.kt

```diff
@@ -1,7 +1,106 @@
+import kotlinx.coroutines.launch
+import kotlin.random.Random
 
-class GeneralContentViewModel : BaseViewModel() {
+class GeneralContentViewModel(private val repo: IntroRepository) : BaseLoadMoreViewModel<Recipe>() {
+    private val _suggestRecipe = MutableLiveData<IntroRecipe>()
+    val suggestRecipe: LiveData<IntroRecipe> get() = _suggestRecipe
+
+    private val _mainRecipes = MutableLiveData<IntroRecipeResponse>()
+    val mainRecipes: LiveData<IntroRecipeResponse> get() = _mainRecipes
+
+    private val _dessertRecipes = MutableLiveData<IntroRecipeResponse>()
+    val dessertRecipes: LiveData<IntroRecipeResponse> get() = _dessertRecipes
+
+    private val _vietnameseRecipes = MutableLiveData<IntroRecipeResponse>()
+    val vietnameseRecipes: LiveData<IntroRecipeResponse> get() = _vietnameseRecipes
+
+    override fun loadData() {
+        viewModelScope.launch {
+            val response = repo.getRandomRecipes(20)
+            if (response.status == Status.SUCCESS) {
+                onLoadSuccess(response.data?.recipes)
+            } else {
+                response.message?.let { onLoadFailed(it) }
+            }
+        }
+    }
+
+    private fun getDessertRecipes() {
+        viewModelScope.launch {
+            val response = repo.searchRandomRecipes(
+                type = TYPE_DESSERT,
+                sort = SORT_RANDOM,
+                sortDirection = DIRECTION_DESC,
+                number = 10
+            )
+            if (response.status == Status.SUCCESS) {
+                _dessertRecipes.value = response.data
+            } else {
+                response.message?.let { setMessage(it) }
+            }
+        }
+    }
+
+    private fun getMainRecipes() {
+        setOnLoading()
+        viewModelScope.launch {
+            val response = repo.searchRandomRecipes(
+                type = TYPE_MAIN,
+                sort = SORT_RANDOM,
+                sortDirection = DIRECTION_DESC,
+                number = 10
+            )
+            if (response.status == Status.SUCCESS) {
+                _mainRecipes.value = response.data
+                val random = Random
+                //Log.d("asd", "main: ${response.data?.introRecipes.size}")
+                _suggestRecipe.value = response.data?.introRecipes?.get(random.nextInt(9))
+                setCloseLoading()
+            } else {
+                response.message?.let { setMessage(it) }
+            }
+        }
+    }
+
+    private fun getVietNameseRecipes() {
+        viewModelScope.launch {
+            val response = repo.searchRandomRecipes(
+                cuisine = CUISINE_VIETNAMESE,
+                sort = SORT_RANDOM,
+                sortDirection = DIRECTION_DESC,
+                number = 10
+            )
+            if (response.status == Status.SUCCESS) {
+                _vietnameseRecipes.value = response.data
+            } else {
+                response.message?.let { setMessage(it) }
```

Comment:
thử tập viết kiểu này cho code gọn hơn nữa nè, nếu ko để nguyên vẫn đc
```suggestion
                response.message?.let(::setMessage)
```